lib/react_client/react_interop.dart

```diff
@@ -34,8 +36,83 @@ abstract class React {
 
   external static bool isValidElement(dynamic object);
   external static ReactClass get Fragment;
+
+  external static JsRef createRef();
+}
+
+/// Creates a [Ref] object that can be attached to an element via the ref attribute.
```

Comment:
"prop" would be more accurate here than "attribute".

Also, #nit the term "element" is a little overloaded, so we prefer distinguishing between DOM Elements and ReactElements. 

```suggestion
/// Creates a [Ref] object that can be attached to a [ReactElement] via the ref prop.
```